spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -55,7 +55,12 @@ case class Etl(args: Seq[String], @transient modules: Seq[TypedModule] = Etl.def
   def scheme: Either[LayoutScheme, LayoutDefinition] = {
     if (conf.layoutScheme.isDefined) {
       val scheme = conf.layoutScheme()(conf.crs(), conf.tileSize())
-      logger.info(scheme.toString)
+      if (conf.maxZoom.isDefined) {
+        scheme match {
+          case  zoomedLayoutScheme: ZoomedLayoutScheme => logger.info("maxZoom and layout scheme compatible")
+          case _ => throw new RuntimeException("maxZoom option should only be used with tms layout scheme")
+        }
+      }
```

Comment:
Thanks @echeipesh,
I've been trying to implement this but can't get it to work properly, I think because layoutScheme is returning a  LayoutSchemaProvider instead of a LayoutScheme (Zoomed or Floating) within the validate function.  Any ideas on how to get the actual LayoutScheme class here?

```
  validateOpt (layoutScheme, maxZoom) {
    case (Some(schema), Some(zoom)) =>  schema match {
        case scheme: ZoomedLayoutScheme => Right(Unit)
        case _ => Left("maxZoom should only be used with ZoomedLayoutScheme (tms)")
      }
    case (None, Some(zoom)) => Left("maxZoom not be used when a LayoutScheme is not provided")
    case _ => Right(Unit)
  }
```